rllib/connectors/tests/test_agent.py

```diff
@@ -450,8 +449,13 @@ def test_vr_connector_with_multiple_buffers(self):
                 act_list.append(act_arrs[t - 1])
 
             self.assertTrue("context_next_obs" not in sample_batch)
+            # We should have the 5 (context_len) most recent observations here
             check(sample_batch["context_obs"], np.stack(obs_list)[None])
-            check(sample_batch["context_act"], np.stack(act_list[:-1])[None])
+            # The context for actions is [t-context_len+1:t]. Since we build sample
+            # batch for inference in ViewRequirementAgentConnector, it always
+            # includes everything up until the last action (at t-1), but not the
+            # action current action (at t).
+            check(sample_batch["context_act"], np.stack(act_list[1:])[None])
```

Comment:
This looked correct before. But the way it was before it never included the action we added before calling build_for_inference. But this is what happens with State Buffer connector.